Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortableModBuilder.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.IO;
+using ModCompendiumLibrary.Logging;
+using ModCompendiumLibrary.VirtualFileSystem;
+using CriPakTools;
+using ModCompendiumLibrary.Configuration;
+using System.Diagnostics;
+
+namespace ModCompendiumLibrary.ModSystem.Builders
+{
+    public abstract class PersonaPortableModBuilder : IModBuilder
+    {
+        protected abstract Game Game { get; }
+
+        public bool OutputUnmodifiedFiles { get; } = true;
+        /// <inheritdoc />
+        public VirtualFileSystemEntry Build(VirtualDirectory root, string hostOutputPath = null, string compression = null)
+        {
+            if (root == null)
+            {
+                throw new ArgumentNullException(nameof(root));
+            }
+
+            // Get game config
+            var config = ConfigManager.Get(Game) as PersonaPortableGameConfig;
+            if (config == null)
+            {
+                // Unlikely
+                throw new InvalidOperationException("Game config is missing.");
+            }
+
+            if (string.IsNullOrWhiteSpace(config.CpkRootOrPath))
+            {
+                throw new InvalidConfigException("CPK path is not specified.");
+            }
+
+            // Create temp folder
+            var cpkRootDirectoryPath = Path.Combine(Path.GetTempPath(), $"{Game}ModCompilerTemp_" + Path.GetRandomFileName());
+            Log.Builder.Trace($"Creating temp output directory: {cpkRootDirectoryPath}");
+            Directory.CreateDirectory(cpkRootDirectoryPath);
+
+            // Get files from CPK
+            VirtualDirectory cpkRootDirectory;
+
+            Log.Builder.Trace($"RootOrCpkPath = {config.CpkRootOrPath}");
```

Comment:
minor nit, but there's a discrepancy between the log text and the actual name of the variable.
you could solve this by using `nameof`